### PR TITLE
Use GenAI instead of LLM and other clean ups

### LINKED ISSUE #969: Update LLM references to "Gen AI" where attributes/spans are not exclusive to LLM requests.
Update tags, names and descriptions in the attributes registry to use gen_ai instead of llm, and Generative AI instead of LLM. For attributes that are only relevant to LLM, the reference can remain.

### PR DESCRIPTION
**Pull request by lmolkova**
Fix #969
With #811 merged, we now have inconsistent LLM vs GenAI usage. This PR:
- uses GenAI instead of LLM
- cleans up unused properties in yaml 

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: model/trace/gen-ai.yaml [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
     brief: >
-      A request to an LLM is modeled as a span in a trace. The span name should be a low cardinality value representing the request made to an LLM, like the name of the API endpoint being called.
+      Describes GenAI operation span.
     attributes:
       - ref: gen_ai.system
         requirement_level: required
-        note: >
-          If not using a vendor-supplied model, provide a custom friendly name, such as a name of the company or project.
-          If the instrumetnation reports any attributes specific to a custom model, the value provided in the `gen_ai.system` SHOULD match the custom attribute namespace segment.
-          For example, if `gen_ai.system` is set to `the_best_llm`, custom attributes should be added in the `gen_ai.the_best_llm.*` namespace.
```
Comment (lmolkova) [inline model/trace/gen-ai.yaml:12]: removed since we don't have any system-specific attributes - we can add similar section back once we have any

**Review by joaopgrassi — APPROVED** [review]

File: docs/gen-ai/gen-ai-spans.md [hunk 2]
```diff
@@ -127,4 +128,4 @@ The event name MUST be `gen_ai.content.completion`.
 <!-- END AUTOGENERATED TEXT -->
 <!-- endsemconv -->
 
-[DocumentStatus]: https://github.com/open-telemetry/opentelemetry-specification/tree/v1.22.0/specification/document-status.md
+[DocumentStatus]: https://github.com/open-telemetry/opentelemetry-specification/tree/v1.33.0/specification/document-status.md
```
[the remark this answers is not in the document]
Comment (lmolkova) [reply docs/gen-ai/gen-ai-spans.md:131]: they are, but only if the version matches the expected one.

E.g. 
`https://github.com/open-telemetry/opentelemetry-specification/tree/v1.33.0/...` is replaced by the script when updating from `1.33.0`. 
`https://github.com/open-telemetry/opentelemetry-specification/tree/anything_else/...` is not which makes sense - there could be a permalink to an older version that we'd want to preserve.

[the remark this answers is not in the document]
Comment (lmolkova) [reply docs/gen-ai/gen-ai-spans.md:131]: I guess we can change all the links to point to otel.io and save ourselves a hustle

[the remark this answers is not in the document]
Comment (lmolkova) [reply docs/gen-ai/gen-ai-spans.md]: Moved all the link updates to https://github.com/open-telemetry/semantic-conventions/pull/1093

**Review by drewby — APPROVED** [review]
LGTM.

File: model/registry/gen-ai.yaml [hunk 3]
```diff
@@ -92,17 +84,15 @@ groups:
       - id: prompt
         stability: experimental
         type: string
-        brief: The full prompt sent to an LLM.
+        brief: The full prompt sent to GenAI model.
```
Comment (TaoChenOSU) [inline model/registry/gen-ai.yaml:87]: nit:
```suggestion
        brief: The full prompt sent to the GenAI model.
```

File: model/registry/gen-ai.yaml [hunk 4]
```diff
@@ -19,62 +18,55 @@ groups:
...
+
+          For custom model, a custom friendly name SHOULD be used.
+          If none of these options apply, the `gen_ai.system` SHOULD be set to `_OTHER`.
         examples: 'openai'
-        tag: llm-generic-request
       - id: request.model
         stability: experimental
         type: string
-        brief: The name of the LLM a request is being made to.
+        brief: The name of the Generative AI model a request is being made to.
```
Comment (TaoChenOSU) [inline model/registry/gen-ai.yaml:28]: nit: for consistency
```suggestion
        brief: The name of the GenAI model a request is being made to.
```

**Review by TaoChenOSU — APPROVED** [review]

File: docs/attributes-registry/gen-ai.md [hunk 5]
```diff
@@ -12,26 +12,27 @@ This document defines the attributes used to describe telemetry in the context o
```
Comment (mattmccleary) [inline docs/attributes-registry/gen-ai.md:12]: Should the sentence prior be updated to read, "This document defines the attributes used to describe telemetry in the context of Generative Artificial Intelligence (GenAI) requests and responses."?

File: docs/attributes-registry/gen-ai.md [hunk 6]
```diff
@@ -12,26 +12,27 @@ This document defines the attributes used to describe telemetry in the context o
...
-| `gen_ai.request.model`           | string   | The name of the LLM a request is being made to.                                                  | `gpt-4`                                                                 | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `gen_ai.request.temperature`     | double   | The temperature setting for the LLM request.                                                     | `0.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `gen_ai.request.top_p`           | double   | The top_p sampling setting for the LLM request.                                                  | `1.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `gen_ai.prompt`                  | string   | The full prompt sent to GenAI model. [2]                                                         | `[{'role': 'user', 'content': 'What is the capital of France?'}]`       | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
Comment (mattmccleary) [inline docs/attributes-registry/gen-ai.md:17]:
```suggestion
| `gen_ai.prompt`                  | string   | The full prompt sent to the GenAI model. [2]                                                         | `[{'role': 'user', 'content': 'What is the capital of France?'}]`       | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

File: docs/attributes-registry/gen-ai.md [hunk 7]
```diff
@@ -12,26 +12,27 @@ This document defines the attributes used to describe telemetry in the context o
...
+| `gen_ai.request.max_tokens`      | int      | The maximum number of tokens the model generates for a request.                                  | `100`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `gen_ai.request.model`           | string   | The name of the Generative AI model a request is being made to.                                  | `gpt-4`                                                                 | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
Comment (mattmccleary) [inline docs/attributes-registry/gen-ai.md:19]:
```suggestion
| `gen_ai.request.model`           | string   | The name of the GenAI model a request is being made to.                                  | `gpt-4`                                                                 | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

**Pull request merged**